src/msg/async/EventKqueue.cc

```diff
@@ -156,8 +156,13 @@ int KqueueDriver::add_event(int fd, int cur_mask, int add_mask)
     }
   }
   // keep what we set
-  if (fd >= sav_max)
-    resize_events(sav_max+5000);
+  if (fd >= sav_max) {
+    r = resize_events(sav_max+5000);
+    if (r < 0) {
+      lderr(cct) << __func__ << " event count is exceed." << dendl;
+      return r;
```

Comment:
i suggested `assert(sav_events)` after the `realloc()` call. and i still do. because we are *not*  able to recover from the failed call of `realloc()` .